### PR TITLE
Apikey authentication query

### PR DESCRIPTION
**Pull request by maggarwal13**

### CODE REVIEW

File: whois-client/src/main/java/net/ripe/db/whois/common/apiKey/OAuthSession.java
Comment (MiguelAHM) [inline whois-client/src/main/java/net/ripe/db/whois/common/apiKey/OAuthSession.java:63]: why this class is not a record?

File: whois-commons/src/main/java/net/ripe/db/whois/common/rpsl/transform/FilterAuthFunction.java
Comment (MiguelAHM) [inline whois-commons/src/main/java/net/ripe/db/whois/common/rpsl/transform/FilterAuthFunction.java:122]: Shouldn't be better to generate another list with mntners? I mean you are affecting rest of authentication methods adding mnt-by attributes
Shouldn't be any side effect because mnt-by format doesn't match the regex in each of them...however I think that it is safer to manage a different list ad-hoc for apikeyAuthenticated
something like:
`
final List<RpslAttribute> extendedAuthAttributes = Lists.newArrayList(rpslObject.findAttributes(AttributeType.AUTH));
`
`
final List<RpslAttribute> apikeyExtendedAttributes =  extendedAuthAttributes.addAll(getMntByAuthAttributes(maintainers));
`
`
apiKeyAuthenticated(rpslObject, maintainers, apikeyExtendedAttributes)
`

File: whois-commons/src/main/java/net/ripe/db/whois/common/rpsl/transform/FilterAuthFunction.java
Comment (MiguelAHM) [inline whois-commons/src/main/java/net/ripe/db/whois/common/rpsl/transform/FilterAuthFunction.java:221]: are we ignoring the fourth part of the scope?
It will always be "write" right now.
Maybe we should check it...just for being consistent?
I don't know, I leave it to you

File: whois-client/src/main/java/net/ripe/db/whois/common/apiKey/OAuthSession.java
Comment (maggarwal13) [reply whois-client/src/main/java/net/ripe/db/whois/common/apiKey/OAuthSession.java:63]: Why you want to make it a record? This will be JSON parsed  once proxy layer is implemented.

File: whois-commons/src/main/java/net/ripe/db/whois/common/rpsl/transform/FilterAuthFunction.java
Comment (maggarwal13) [reply whois-commons/src/main/java/net/ripe/db/whois/common/rpsl/transform/FilterAuthFunction.java:221]: Yes , it will not work as I need to add the dependency whois-update on whois-query which I don't want and this does not belong to whois-common.I am not sure, I can move the whole APIKeyValidator to whois common but that will also not work as there are lot of implementation of that interface so need to do lot of refactoring. We are doing the same thing ClientAuthCertificateValidator so I just did the same for consitency.

Comment (maggarwal13) [reply whois-commons/src/main/java/net/ripe/db/whois/common/rpsl/transform/FilterAuthFunction.java:221]: May be I can do other way around. Let me try

File: whois-client/src/main/java/net/ripe/db/whois/common/apiKey/OAuthSession.java
Comment (MiguelAHM) [reply whois-client/src/main/java/net/ripe/db/whois/common/apiKey/OAuthSession.java:63]: no, I don't care if it is a record or not.
Just wander why it is no a record...

File: whois-commons/src/main/java/net/ripe/db/whois/common/rpsl/transform/FilterAuthFunction.java
Comment (MiguelAHM) [reply whois-commons/src/main/java/net/ripe/db/whois/common/rpsl/transform/FilterAuthFunction.java]: Thanks for the change

**Review by MiguelAHM — APPROVED** [review]

**Pull request merged**
